x-pack/plugins/security_solution/public/detections/components/rules/query_bar/index.tsx

```diff
@@ -137,7 +163,28 @@ export const QueryBarDefineRule = ({
     return () => {
       isSubscribed = false;
     };
-  }, [fieldValue, filterManager, savedQuery, savedQueryServices]);
+  }, [
+    fieldValue,
+    filterManager,
+    savedQuery,
+    savedQueryServices,
+    setIsSavedQueryFailedToLoad,
+    setFieldValue,
+  ]);
+
+  useEffect(() => {
+    if (isSavedQueryFailedToLoad) {
+      onSavedQueryError?.();
```

Comment:
Nit: While straightforward for statements like `const x = a?.b?.c ?? 'default value';`, this syntax always makes me scratch my head for 1 or 2 seconds when used for conditional function calls. I'd use a simple `if` in this case :)

Reply:
I like it as it's more laconic and readable approach. Also, fairly frequently used in codebase as well